File: tarea 3/En Python/ejercicio_7.py
x = float(input("Ingrese un número positivo: "))

error = 0.000001

if x < 0:
    print("No existe raíz cuadrada real para números negativos.")
elif x == 0:
    print("La raíz cuadrada es: 0")
else:
    s = x

    while abs(s * s - x) > error:
        s = 0.5 * (s + x / s)

    print("La raíz cuadrada aproximada es:", s)

File: tarea 3/En C++/ejercicio 7.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    double x, s, error = 0.000001;

    cout << "Ingrese un numero positivo: ";
    cin >> x;

    if (x < 0) {
        cout << "No existe raiz cuadrada real para numeros negativos." << endl;
        return 0;
    }

    if (x == 0) {
        cout << "La raiz cuadrada es: 0" << endl;
        return 0;
    }

    s = x; // Aproximacion inicial

    while (fabs(s * s - x) > error) {
        s = 0.5 * (s + x / s);
    }

    cout << "La raiz cuadrada aproximada de " << x << " es: " << s << endl;

    return 0;
}
